hail/python/hailtop/hailctl/batch/batch_cli_utils.py

```diff
@@ -35,32 +36,32 @@ def bool_string_to_bool(bool_string):
     raise ValueError("Input could not be resolved to a bool")
 
 
-def separated_format(lod: LoD, delim: str) -> None:
-    writer = csv.writer(sys.stdout, delimiter=delim)
-    # construct header dictionary from keys of lod
-    header = {k: k for (k, v) in lod[0].items()}
+def separated_format(table_data: TableData, delim: str) -> str:
+    output = io.StringIO()
+    writer = csv.writer(output, delimiter=delim)
+    header = {k: k for k in table_data[0].keys()}
     writer.writerow(header)
-    # for each of the elements in the list of dicts, write its values as a list to the writer
-    for d in lod:
-        writer.writerow([v for (k, v) in d.items()])
+    for d in table_data:
+        writer.writerow([v for v in d.values()])
 
+    contents = output.getvalue()
+    output.close()
+    return contents
 
-def table_format(lod: LoD, tablefmt: str) -> None:
-    print(tabulate.tabulate(lod, headers='keys', tablefmt=tablefmt))
 
+def table_format(table_data: TableData, tablefmt: str) -> str:
+    return tabulate.tabulate(table_data, headers='keys', tablefmt=tablefmt)
 
+
+# PRE-CONDITION: name is a valid option for the -o argument of the CLI
```

Comment:
Assertions in code are preferred over those in comments. Instead, let's add an assert that `name` is in our choices array at the begginning of `make_formatter`.